### PR TITLE
Create convention file to follow in all contributions

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changes description 🧑‍💻

[chore: Update Compose, Kotlin and AGP dependencies.](https://github.com/Decathlon/vitamin-compose/commit/4b2b47c66329e3931945fde4e2bac7ab1cf28a06)
 
[refactor(foundation): Move ripple to the Vitamin theme](https://github.com/Decathlon/vitamin-compose/commit/85bbd0979a89c524ce98eb4902067f697d3c861d)

[refactor(buttons): Migrate to the new composable contract convention](https://github.com/Decathlon/vitamin-compose/commit/ad64e3a8f0d2585c36282e6db6ff756629b23807)

[docs: Add conventions about components.](https://github.com/Decathlon/vitamin-compose/commit/c50263046a9be7bd5218d7378a349e6890a0abd0)

## Context 🤔

Not linked to an issue

## Checklist ✅

- [x] I have reviewed the submitted code
- [x] I have tested on a phone device/emulator

### CODE REVIEW

**Comment by dktbitrise** [comment]
### You can download the apk here :

**Review by adrienbusin — APPROVED** [review]

File: CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        ripple: RippleTheme = VitaminTheme.ripples.brand,
+        onClick: () -> Unit
+    )
+}
+```
+
+## Api Slotting convention
+
+According to the design specification, some component can have multiple variant in one of the 
+internal component used. If it is the case, the api slotting should be used to purpose the best
```
Comment (FDELAHA24) [inline CONVENTIONS.md:95]:
```suggestion
internal component used. If it is the case, the api slotting should be used to offer the best
```

**Review by CorentinStamperDecathlon — APPROVED** [review]

**Review by FDELAHA24 — CHANGES_REQUESTED** [review]
Good for me, just very few documentation changes suggestions, Good job !

File: CONVENTIONS.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+If your component can be customized by properties, you must create an object per property with the
+following specification for the name `Vitamin<Component><Property>s`. e.g. `VitaminButtonColors`
+
+If you component have only one variant and if there isn't any name proposed by the specification
+in the Figma project, the composable inside the object access must be named `Primary`. e.g. for
+a component with only one variant `VitaminRadioButtons.Primary`.
+
+When you create a common component for all your variant, it must be `private` and with the
+following specification for the name `Vitamin<Component><Variant>?Impl`. e.g. `VitaminButtonImpl`
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply CONVENTIONS.md:25]: Maybe "Variant" isn't the right name? Maybe something like "Domain" which can mean: "If you component can cover several domains, you can create a common component by domain with the follow specification ..."

[the remark this answers is not in the document]
Comment (GerardPaligot) [reply CONVENTIONS.md:25]: e.g: in the appbar, we have the TopAppBar and the BottomAppBar. Of course, you won't have the same common component.

File: CONVENTIONS.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    fun Primary(
+        text: String,
+        modifier: Modifier = Modifier,
+        icon: Painter? = null,
+        iconSide: IconSide = IconSide.LEFT,
+        enabled: Boolean = true,
+        colors: ButtonColors = VitaminButtonsColors.primary(),
+        sizes: VitaminButtonSizes = VitaminButtonsSizes.mediumSize(),
+        ripple: RippleTheme = VitaminTheme.ripples.brand,
+        onClick: () -> Unit
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply CONVENTIONS.md:87]: This allows you to have this king of usage:
```kotlin
MyComponent {
  // My last parameter is a callback so I can do this kind of thing
}
```

[the remark this answers is not in the document]
Comment (ManonPolle) [reply CONVENTIONS.md:87]: 👍

**Review by FDELAHA24 — APPROVED** [review]

**Comment by dktbitrise** [comment]
### You can download the apk here :
**sample-debug.apk :** 
https://app.bitrise.io/app/e9ae480151be4514/build/1456ea14-0cd6-4f27-b49b-a228c769a44d/artifact/02856e3657f54c3f/p/3a9932ad51f2d3250782f2ca7f6888fa

**Review by ManonPolle — APPROVED** [review]

**Pull request merged**